Chessington.GameEngine/Pieces/King.cs

```diff
@@ -10,7 +10,27 @@ public King(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            List<Square> availableMoves = new List<Square>();
+            
+            int[] row = new int[] { 1, 1, 1, 0, 0, -1, -1, -1};
+            int[] col = new int[] { -1, 0, 1, -1, 1, -1, 0, 1};
+            
+            for (int i = 0; i < 8; i++)
+            {
+                if(board.FindPiece(this).Row + row[i] >= 0
```

Comment:
board.FindPiece(this) could be replaced with a variable above to avoid multiple function calls

Reply:
this is also true for the other classes

Reply:
Yessir that is true. Would've saved me some time and space